echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/artifacts/ArtifactMatcher.java

```diff
@@ -86,6 +93,55 @@ public static boolean isConstraintInPayload(final Map constraints, final Map pay
     return true;
   }
 
+  /**
+   * Check that there is a key in the payload for each constraint declared in a Trigger. Also check
+   * that if there is a value for a given key, that the value matches the value in the payload.
+   *
+   * <p>The constraint key may accept a JsonPath expression for deeper json search, the evaluation
+   * value should be a String or a List<String>.
+   *
+   * @param constraints A map of constraints configured in the Trigger (eg, created in Deck). A
+   *     constraint is a [key, java regex value] pair or a [JsonPathExp, java regex value].
+   * @param payload A map of the payload contents POST'd in the triggering event.
+   * @return Whether every key or expression (and value if applicable) in the constraints map is
+   *     represented in the payload.
+   */
+  public static boolean isJsonPathConstraintInPayload(final Map constraints, final Map payload) {
+    String json = gson.toJson(payload);
+    DocumentContext documentContext = JsonPath.using(conf).parse(json);
+
+    for (Object key : constraints.keySet()) {
+      if (!payload.containsKey(key) || payload.get(key) == null) {
+        String value;
+        try {
+          value = documentContext.read(key.toString(), String.class);
+        } catch (InvalidPathException e) {
```

Comment:
Is this exception thrown when the provided jsonpath is invalid? It might be nice to add validation at save time to ensure that this doesn't actually happen in practice. It also might be nice to at least log the error.

Reply:
Yes, it is for invalida jsonpath, if the exception is thrown it logs the error. The validation would be nice but I think it should be in a different PR.